ci/compute-matrix.jq

```diff
@@ -23,6 +23,16 @@ def compute_tag_prefix($x):
     $x.IMAGE_REPO + "-" + env.PR_NUM + "-" + env.RAPIDS_VERSION_MAJOR_MINOR + "-"
   end;
 
+def compute_tag_prefix_no_rapids_version($x):
+  if
+    env.BUILD_TYPE == "branch"
+  then
+    ""
+  else
+    $x.IMAGE_REPO + "-" + env.PR_NUM + "-"
+  end;
```

Comment:
Duplicating these functions with `_no_rapids_version` forms is not the most elegant way to do this, but it was easy to write (for me, someone not that familiar with `jq`) and I think it's easy to read. Happy to try a different way if reviewers are uncomfortable with this logic being duplicated.